### ISSUE TITLE
Unable to decrease loss in 20000 steps

### ISSUE THREAD

**Issue by prakhar2811** [tensorflow/ranking#70]
Should i try feature engineering, normalizing features etc?

**Comment by HongleiZhuang** [comment]
There are many possible issues and you may want to check the followings before you go for feature engineering.  

First, does the loss remains exactly the same throughout 20,000 steps without even a little bit of change?  If so, there is likely something wrong with the data.  Check if you properly specify the data paths, organizing them into the correct tensor shapes, etc.  

If the loss fluctuates but does not decrease significantly, then something might be wrong with the optimization part.  A common issue is the learning rate being too small or too large.  You may try to explore learning rates at different magnitudes.  For example, if you are using learning rate of 1e-5, you may try 1e-4, 1e-3, or 1e-6, 1e-7 etc.  Trying different optimizer may also help.  

There could be other possible issues but I would need more information before I can help you with the diagnosis.  If possible, can you provide more details about how you train the model?  For example, what optimizer do you use?  What is the learning rate?  What is the loss function?  What is the size of your data set? etc.  

Thank you.
